service/lib/agama/manager.rb

```diff
@@ -102,7 +102,10 @@ def config_phase
       #   storage config could be applied before probing. In that case, the config has to be
       #   recovered.
       progress.step { first_time ? probe_and_recover_storage : storage.probe }
-      progress.step { software.probe }
+      progress.step do
+        software.probe
+        security.probe
```

Comment:
I miss comment here we discuss about reasons why security have to be after software, so we do not break it in future ( or can re evaluate if conditions change )